cql3/query_processor.cc

```diff
@@ -1035,9 +1035,16 @@ query_processor::execute_schema_statement(const statements::schema_altering_stat
     }
 
     ce = std::move(ret);
-
-    // If an IF [NOT] EXISTS clause was used, this may not result in an actual schema change.  To avoid doing
-    // extra work in the drivers to handle schema changes, we return an empty message in this case. (CASSANDRA-7600)
+    if (!ce) {
+        // If an IF [NOT] EXISTS clause was used and resource was already created
+        // we don't emit schema change event. However it interact badly with
+        // concurrent clients creating resources. The client seeing no create event
+        // assumes resource is created and proceeds with its logic which may depend on
```

Comment:
here the clearer tense is not "is created" (which may make it sound like it was created right now) but maybe "already previously existed" or something?

Reply:
fixed